ddtrace/contrib/flask/patch.py

```diff
@@ -279,6 +282,42 @@ def traced_start_response(status_code, headers):
     return traced_start_response
 
 
+@with_instance_pin
+def traced_stream_with_context(pin, wrapped, instance, args, kwargs):
+    """
+    Wrapper for flask.stream_with_context
+
+    This function wraps streamed responses.
+    Note - Responses are stream after the flask wsgi app is called. Therefore the span produced by this
+    wrapper will be a sibling span of flask.request.
+    """
+    # stream_with_context(...) copies the current request context before returning an iterable.
+    # To avoid raising a RuntimeError in the wsgi server we must wrap the iterable returned by
+    # this function: https://github.com/pallets/flask/blob/2.1.2/src/flask/helpers.py#L113
+    iterable = wrapped(*args, **kwargs)
+
+    resource = u" ".join((flask.request.method, flask.request.path))
+
+    def _wraped_generator():
+        # TO FIX: GeneratorExit is called when the generator returned stream_with_context(...) is garbage collected.
+        # Calling span.finish() at the point might work in CPython but this could cause issues in other python
+        # implementation (ex. PyPy) or when a reference to the generator persists well after the final iteration.
+        span = pin.tracer.trace(
+            "flask.streamed_response",
+            service=trace_utils.int_service(pin, config.flask),
+            resource=resource,
+            span_type=SpanTypes.WEB,
+        )
+        while True:
+            try:
+                yield iterable.__next__()
+            except GeneratorExit:
+                span.finish()
+                raise
```

Comment:
```suggestion
        with pin.tracer.trace(
            "flask.streamed_response",
            service=trace_utils.int_service(pin, config.flask),
            resource=resource,
            span_type=SpanTypes.WEB,
        ):
            while True:
                yield iterable.__next__()
```


Does something like this work?

Or well, we probably need to ignore `StopIteration` if we do this?

or maybe:

```python
span = pin.tracer.trace(...)
try:
    while True:
        yield iterable.__next__()
finally:
    span.finish()
```

Which would ignore any exceptions raised, but is that what we'd want? 🤔

Reply:
Both would work. I just went with the first approach to catch Exceptions